org-code-javabuilder/lib/src/main/java/org/code/neighborhood/GridFactory.java

```diff
@@ -0,0 +1,66 @@
+package org.code.neighborhood;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+import java.io.File;
+import java.io.FileInputStream;
+
+public class GridFactory {
+    protected Grid createGridFromJSON(String filename) {
+        File file = new File("grid.txt");
+        FileInputStream fis;
+        try {
+            fis = new FileInputStream(file);
+            byte[] data = new byte[(int) file.length()];
+            fis.read(data);
+            fis.close();
+            return createGridFromString(new String(data, "UTF-8"));
+        } catch (Exception e) {
+            throw new UnsupportedOperationException("Try adding a grid.txt file.");
+        }
+    }
+
+    protected Grid createGridFromString(String description) {
+        String[] lines = description.split("\n");
+        String parseLine = String.join("", lines);
+        JSONParser jsonParser = new JSONParser();
+        try {
+            Object obj = jsonParser.parse(parseLine);
+            JSONArray gridSquares = (JSONArray) obj;
+            int height = gridSquares.size();
+            if(height == 0) {
+                throw new UnsupportedOperationException("Please check the format of your grid.txt file");
+            }
+            int width = ((JSONArray) gridSquares.get(0)).size();
+            GridSquare[][] grid = new GridSquare[width][height];
+            int currentHeight = height;
+            for (int currentY = 0; currentY < height; currentY++) {
+                currentHeight--;
```

Comment:
can you add some comments in here about the expected format of the `description` string and why we are starting from the maxHeight instead of 0?